js/main.js

```diff
@@ -1,16 +1,101 @@
+];
+const ADS_FEATURES = ['wifi', 'dishwasher', 'parking', 'elevator', 'conditioner'];
+const ADS_DISCRIPTION = [
+  'Все окна на север',
+  'Тихие соседи',
+  'Южная сторона',
+  'Панорамные окна'
+];
+const ADS_PHOTOS = [
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/duonguyen-8LrGtIxxa4w.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/brandon-hoogenboom-SNxQGWxZQi0.jpg',
+  'https://assets.htmlacademy.ru/content/intensive/javascript-1/keksobooking/claire-rendall-b6kAwr1i0Iw.jpg'
+];
+
+const getRandomPositiveInteger = (a, b) => {
+  const lower = Math.ceil(Math.min(Math.abs(a), Math.abs(b)));
+  const upper = Math.floor(Math.max(Math.abs(a), Math.abs(b)));
+  const result = Math.random() * (upper - lower + 1) + lower;
+  return Math.floor(result);
+};
+
+const getRandomPositiveFloat = (a, b, digits = 1) => {
+  const lower = Math.min(Math.abs(a), Math.abs(b));
+  const upper = Math.max(Math.abs(a), Math.abs(b));
+  const result = Math.random() * (upper - lower) + lower;
+  return +result.toFixed(digits);
 };
 
-getRandomInt(1, 1);
+const getRandomArrayElement = (elements) => elements[getRandomPositiveInteger(0, elements.length - 1)];
+
+const createRandomArray = (values) => {
+  const features = new Array(getRandomPositiveInteger(1, values.length));
+  for (let i = 0; i < features.length; i++) {features[i] = values[i];}
+  return features;
+};
+
+const createAutor = (avatarNumber) => ({
+  avatar: `img/avatars/user${avatarNumber}.png`,
+});
+
+const createOffer = (lat, lng) => ({
+  title: getRandomArrayElement(ADS_TITLES),
+  address: `${lat} ${lng}`,
+  price: getRandomPositiveInteger(1000, 5000),
+  type: getRandomArrayElement(ADS_TYPES),
+  rooms: getRandomPositiveInteger(1, 5),
+  guests: getRandomPositiveInteger(1, 10),
+  checkin: getRandomArrayElement(CHECK_IN),
+  checkout: getRandomArrayElement(CHECK_OUT),
+  features: createRandomArray(ADS_FEATURES),
+  description: getRandomArrayElement(ADS_DISCRIPTION),
+  photos: createRandomArray(ADS_PHOTOS),
+});
+
+const createLocation = (lat, lng) => ({
+  lat,
+  lng,
+});
+
+const createAd = (avatarNumber) => {
+  const lat = getRandomPositiveFloat(35.65000, 35.70000, 5);
+  const lng = getRandomPositiveFloat(139.70000, 139.80000, 5);
+  return {
+    author: createAutor(avatarNumber),
+    location: createLocation(lat, lng),
+    offer: createOffer(lat, lng),
+  };
+};
 
-const getRandomGeo = (firstValue, secondValue, num) => {
-  if (firstValue < 0 || secondValue < 0 || (secondValue - firstValue) < 0) {return null;}
-  if (firstValue.toFixed(num) === secondValue.toFixed(num)) {return firstValue.toFixed(num);}
-  return ((Math.random() * (secondValue - firstValue)) + firstValue).toFixed(num);
+const createAds = () => {
+  const massiveAds = [];
+  for (let i = 1; i <= ADS_NUMBERS; i++) {
+    const avatarNumber = (i < 10) ? `0${i}` : `${i}`;
+    massiveAds.push(createAd(avatarNumber));
+  }
+  return massiveAds;
 };
```

Comment:
```suggestion
const createAds = (count) => {
  const massiveAds = [];
  for (let i = 1; i <= count; i++) {
    const avatarNumber = (i < 10) ? `0${i}` : `${i}`;
    massiveAds.push(createAd(avatarNumber));
  }
  return massiveAds;
};

createAds(ADS_NUMBERS)
```
вот так вот вообще будет красиво!

а вот это вот - уровень Бог (как показывали на лекции):

```ts
const createAds = (count) => {
  const ads = new Array({ lenght: count } , (_, i) => {
    const avatarNumber = (i < 10) ? `0${i}` : `${i}`;
    const ad = createAd(avatarNumber)
    
   return ad
  });

  return ads;
};

createAds(ADS_NUMBERS)
```